.github/workflows/windows.yml

```diff
@@ -0,0 +1,67 @@
+name: Windows Build
+on:
+  push:
+    branches:
+      - 'master'
+      - 'windows-build'
+    paths:
+      - '**.[chly]'
+      - '**/Makefile.am'
+      - '.github/workflows/windows.yml'
+  pull_request:
+    paths:
+      - '**.[chly]'
+      - '**/Makefile.am'
+      - '.github/workflows/windows.yml'
+jobs:
+  windows:
+    runs-on: windows-latest
+    env:
+      APPVER: '10.0'
+      CODESIGN_PKT: 0000000000000000
+      INSTALL_DIR: C:\jq
+      WINSDKVER: '10.0.22000.0'
+      WIXDIR: 'c:\Program Files (x86)\Windows Installer XML v3.5'
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+      - uses: cygwin/cygwin-install-action@v4
+        with:
+          packages: >-
+            coreutils
+            bison
+            autoconf
+            automake
+      # - uses: msys2/setup-msys2@v2
+      #   with:
+      #     update: true
+      #     install: >-
+      #       base-devel git
+      #- name: Set up MinGW
+      #  uses: egor-tensin/setup-mingw@v2
+      #  with:
+      #    platform: x64
+      - name: Build and Test
+        shell: bash
+        run: |
+          git clone https://github.com/jqlang/jq
+          cd jq
```

Comment:
wonder if you can skip this and relay on actions/checkout step above? and i would guess the rest of the steps will run with current working directory being the checkout repo